systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -312,4 +313,28 @@ void testCruiseControlIntraBrokerBalancingWithoutSpecifyingJBODStorage(Extension
         KafkaRebalanceStatus kafkaRebalanceStatus = KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus();
         assertThat(kafkaRebalanceStatus.getConditions().get(0).getReason(), is("InvalidResourceException"));
     }
+
+    @IsolatedTest
+        void testCruiseControlDisabledToEnabled(ExtensionContext extensionContext) {
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3, 3)
+                .editMetadata()
+                .withNamespace(namespace)
+                .endMetadata()
+                .build());
+        resourceManager.createResource(extensionContext, false,  KafkaRebalanceTemplates.kafkaRebalance(clusterName)
+                .editMetadata()
+                .withNamespace(namespace)
+                .endMetadata()
+                .build());
+
+        KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(namespace, clusterName, KafkaRebalanceState.NotReady);
+
+        //Cruise Control spec is now enabled
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, kafka -> kafka.getSpec().setCruiseControl(new CruiseControlSpec()), namespace);
+
+        // The resource will move from `Not Ready` to `ProposalReady`
+        KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(namespace, clusterName, KafkaRebalanceState.ProposalReady);
+    }
```

Comment:
TBH, maybe you can put it into the `testCruiseControlWithRebalanceResourceAndRefreshAnnotation` -> just change the test a bit to save some time. 

Instead of how it is done now, you can just deploy Kafka without CC, deploy Rebalance, wait for `NotReady` state, put the CC to Kafka -> else should be same as it is now.

Reply:
Could it be even faster if we created the Rebalance resource first and then the Kafka cluster with CC enabled? Would this work?

Reply:
@kyguy good idea, it should work similar way, or?

Reply:
@im-konge This way (assuming we can create a KR resource before a existing Kafka cluster without problems) we save ourselves from having the brokers to be rolled to include the CC metric reporters. I'll leave that up to @ShubhamRwt to decide.

Reply:
@kyguy @im-konge If I remember correctly, if we apply a KR resource before a Kafka resource or cluster present, then probably we go into the `NotReady` state and stay there. Until or unless we apply a refresh annotation

Reply:
And the behavior I want to check here is that if cc is not enabled in Kafka resource then once it is enabled the resource will itself move into `Ready` state without the need of `refresh` annotation

Reply:
In both cases you don't have CC deployed, so I thought that it is looking for CC, not for Kafka cluster also. Anyway, you can still do the thing which I proposed above

Reply:
> If I remember correctly, if we apply a KR resource before a Kafka resource or cluster present, then probably we go into the NotReady state and stay there. Until or unless we apply a refresh annotation

@ShubhamRwt  not sure what happens with the reconciliation when a topic or user CR is created before the Kafka cluster is deployed but I imagine we would want consistent behavior for the KafkaRebalance resource. Let me know what you think! 

As for what the system test tests, I'll leave that too you!